source-mysql/prerequisites.go

```diff
@@ -19,7 +25,14 @@ func (db *mysqlDatabase) SetupPrerequisites(ctx context.Context) []error {
 }
 
 func (db *mysqlDatabase) prerequisiteBinlogFormat(ctx context.Context) error {
-	// TODO: Verify that 'binlog_format' is set to 'ROW'
+	var results, err = db.conn.Execute(`SELECT @@GLOBAL.binlog_format;`)
+	if err != nil || len(results.Values) == 0 {
+		return fmt.Errorf("unable to query 'binlog_format' system variable: %w", err)
```

Comment:
If `len(results.Values) == 0`, it seems like the returned error will not be clear if `err == nil`.

Reply:
Good point. I'm _fairly_ sure that that should never happen on any version of MySQL that we care about supporting but I may as well separate that out so we produce a cleaner error message.